### PR TITLE
[CI] Add Dockerfile for setting up CI runner

### PR DESCRIPTION
**Pull request by goldsteinn**
This will make it easier to add future runners.

Wait on success of:
https://github.com/dtcxzyw/llvm-opt-benchmark/actions/runs/11108241699/job/30860722989?pr=1400

to make sure this is correct.

### CODE REVIEW

File: runner-setup/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,63 @@
+# Dockerfile to setup a new runner for CI service.
+# This file has all dependencies for running the following service(s):
+#   - gen_optimized.py
+
+# Use a lightweight base image
+FROM ubuntu:24.04
+
+# Set environment variables
+# Should probably use docker compose for this
+ENV RUNNER_USER=runner
```
Comment (dtcxzyw) [inline runner-setup/Dockerfile:10]: Can we use `ARG` instead? See https://docs.docker.com/reference/dockerfile/#arg.

File: runner-setup/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+ENV RUNNER_HOME=/home/${RUNNER_USER}
+# Should probably use docker compose and to set these values:
+ENV RUNNER_VERSION=2.319.1
+ENV GITHUB_TOKEN=#<Github Token Here>
```
Comment (dtcxzyw) [inline runner-setup/Dockerfile:14]: Use `RUN --mount=type=secret` for secrets: https://docs.docker.com/reference/dockerfile/#example-mount-as-environment-variable

File: runner-setup/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    libssl-dev \
+    libicu-dev \
+    cargo \
+    gawk \
+    bison \
+    wget \
+    flex \
+    curl \
+    jq \
+    git \
```
Comment (dtcxzyw) [inline runner-setup/Dockerfile:37]: Do we have `perf` installed in the base image?

File: runner-setup/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+RUN curl -o actions-runner-linux-x64.tar.gz -L https://github.com/actions/runner/releases/download/v${RUNNER_VERSION}/actions-runner-linux-x64-${RUNNER_VERSION}.tar.gz \
+    && tar xzf ./actions-runner-linux-x64.tar.gz \
+    && rm -f ./actions-runner-linux-x64.tar.gz
+
+# Install the runner
+# We install dependencies manually above
+#RUN ./bin/installdependencies.sh
+
+# Install python3 packages
+RUN pip3 install tqdm --break-system-packages
```
Comment (dtcxzyw) [inline runner-setup/Dockerfile:57]: We can install `python3-tqdm` with `apt`.

File: runner-setup/Dockerfile (hunk 3 above)
Comment (goldsteinn) [reply runner-setup/Dockerfile:37]: Shoot, probably not. Will need to restart.

File: runner-setup/Dockerfile (hunk 1 above)
Comment (goldsteinn) [reply runner-setup/Dockerfile:10]: Yes, ill update  as we will need to restart anyways due to missing perf.

File: runner-setup/Dockerfile (hunk 3 above)
Comment (goldsteinn) [reply runner-setup/Dockerfile:37]: Will probably also need to enable userland events collection.

File: runner-setup/Dockerfile (hunk 2 above)
Comment (goldsteinn) [reply runner-setup/Dockerfile:14]: I think just `ARG` will work here.

Comment (goldsteinn) [reply runner-setup/Dockerfile]: Did secret

**Comment by goldsteinn** [comment]
Updated documentation with everything thats been necessary. Think should be easily reproducible anywhere.

**Review by dtcxzyw — APPROVED** [review]
LGTM. Thank you!

**Pull request merged**